### ISSUE TITLE
Problems with list UI: duplicates

### ISSUE THREAD

**Issue by chepec** [cdevroe/unmark#207]
First, a big thank you to all involved in this project. Very nice to be able to save links for later on a self-hosted service. I've been using it for the last couple of months, and one problem stands out.

For no apparent reason, several links appear twice (at least) in the main list, although I definitely added them only once (using the Chrome extension). Finding the duplicates is made all the more tricky because the in-browser search is ineffective (see #206) and unmark's search only returns one result even though the link is visible more than once in the main list (never immediately next to itself, but a good scroll-distance away).

Has anyone else experienced this duplication issue? It also seems that archiving then deleting a duplicated link removes *all* of them instead of de-duplicating. Quite annoying. I have not been able to find any pattern to which links are duplicated (tags? the way they are added? I'm not sure).

I am using the Chrome "Save to unmark" addon (edited so it points to my self-hosted instance) with the latest stable Chrome version. (Tangential issue: on the [Chrome extension's wiki](https://github.com/plainmade/unmark/wiki/Chrome-Extension), it says search should be possible by typing `nilai` in the omnibar followed by space and the search term. Is this supposed to work? Never worked for me...)

Thanks for taking the time to read this far. I'd be happy to share more technical details if anyone feels this issue is worth investigating.

**Comment by virtadpt** [comment]
I encounter the duplication issue quite frequently - almost every time I acces my Unmark installation to find something.  Most of the time I encounter it in the context of scrolling through my timeline of links - after 

I don't use the Chrome extension because it can't be retargeted without recompiling it.  I use bookmarklets exclusively to store links.  Going through the database dumps, I see no evidence of duplication.

It doesn't seem to be a matter of duplication of links, but of duplication of display of links.  It also seems to happen (on my installation) after I've run a search or two, or after I've archived a couple of links.

**Comment by cdevroe** [comment]
@chepec @virtadpt Do you only find duplicates using the omnibox search feature in the Chrome Extension? Or, do you see duplicates within Unmark app?

Also, if you could get a duplicate to happen and then export your marks into a JSON file and either send them to me privately or put them here I'll be able to troubleshoot the issue.

Site note: I've used Unmark for years and have never seen a duplicate mark added. So I'd happily like to squash this bug for you.

**Comment by cdevroe** [comment]
@chepec @virtadpt Oh, also, please provide version of Unmark you're running.

**Comment by chepec** [comment]
On the server I'm running Unmark 1.7.1.2 (the latest release). In my browser (Chrome, latest stable) I'm running the Unmark extension v1.0.1 (which I use for adding new marks). My desktop runs Debian 8.7 with Gnome. I can observe the same duplicates (as far as I can tell) in both Chrome and Firefox.

I can see multiple cases of duplicates in my list of entries in the Unmark app. I exported all entries as JSON from the settings menu, and interestingly enough, the JSON file shows *no* duplicates (but other than that everything seems to be there).

Also, the Unmark search function always returns one result, and one result only, even for those entries that appear duplicated in the main list.

And finally, I've tested to delete (archive then delete) one of the duplicated entries in the Unmark app, and this removes all instances of it from the main list.

To me these observations indicate that somehow the *display* of the entries becomes duplicated, although the entries in the Unmark database (as reflected in the JSON export) are not.

I hope this makes sense. I'd be happy to email you JSON exports and/or SQL dumps from the server if you think that helps.

**Comment by virtadpt** [comment]
On my server I'm running v1.7.1.2, on top of PHP v5.6 and MySQL v5.6.34.

I do not use the Chrome extension; I do not use Chrome or Chromium, and retargeting the extension to my instance would require a rebuild, which I do not have the skill to do at present.

I see the duplicates when I use the app itself through my web browser.

The next time I see a duplicate I'll export my bookmarks.  Where could I put them privately for you?

**Comment by cdevroe** [comment]
@chepec @virtadpt Is this still an issue in 1.7.2? Please let us know. Also, if you could, look at the database directly to see if the marks seem to be duplicated.

I wish I would replicate this issue but I've simply never seen it.

**Comment by virtadpt** [comment]
Will give it a try and advise.

**Comment by chepec** [comment]
Thank you so much for your efforts. I upgraded to the latest release, but unfortunately I still see duplicates of some entries in the `/marks` list. I also noticed that those entries also show with duplicates in their tag listing (for example the entry `depsy` tagged with `pec` shows up multiple times both in `/marks` and in `/marks/tag/pec`). 

And I checked the SQL database `marks` directly. **220** entries. I checked for duplicate URLs (programmatically) and found none. Also dumped into spreadsheet and checked. No duplicates at all in this table.

Finally, I actually did a manual count of the number of entries in the web UI by simply scrolling slowly and counting. **271** entries. So something is clearly afoot. I'd be happy to run some tests, if you have any suggestions. But I understand you've reworked the UI for the next release, so perhaps this will resolve itself.

**Comment by cdevroe** [comment]
@chepec Looks like a mySQL version issue. Can you confirm your mySQL version please?

**Comment by chepec** [comment]
```
$ mysql --version
mysql  Ver 14.14 Distrib 5.5.57, for debian-linux-gnu (x86_64) using readline 6.3
```

This is on Ubuntu 14.04.5 LTS, running  Apache 2.4.7 and PHP 5.5.9.

**Comment by cdevroe** [comment]
@chepec Do you still use Unmark? If so, do you still see this issue? I'd like to make sure we get this fixed.

**Comment by chepec** [comment]
I'm sorry, I can't be of any help anymore. I moved to another self-hosted bookmarking solution.
I'm sure your efforts will be appreciated by all concerned :-)
